src/irr/asset/CGLIWriter.cpp

```diff
@@ -186,13 +195,18 @@ namespace irr
 					const auto layerData = texture.data(gliLayer, gliFace, region->imageSubresource.mipLevel);
 					const auto sourceData = ptrBeginningOfRegion + (layer * layerSize);
 
-					for (uint32_t yPos = 0; yPos < textureGliImgHeight; ++yPos)
+					for (size_t index = 0; index < pixelOrBlockCount; ++index)
```

Comment:
you need 2 separate loops, one for Z and one for Y,

right now you're copying a whoe image row.line for every block/pixel causes huge overwrites.